### ISSUE TITLE
Set PILOT_ENABLE_METADATA_EXCHANGE to false does not remove tcp MX filter in 1.12

### ISSUE THREAD

**Issue by weibohe** [istio/istio#38520]
### Bug Description

We set `PILOT_ENABLE_METADATA_EXCHANGE` to `false` to disable metadata exchange filters.
We recently upgraded to 1.12 and found out that the flag is not correctly removing TCP metadata exchange filter.

See a sample config dump 
```
         "filter_chain_match": {
          "destination_port": 8080,
          "transport_protocol": "tls"
         },
         "filters": [
          {
           "name": "istio.metadata_exchange",
           "typed_config": {
            "@type": "type.googleapis.com/envoy.tcp.metadataexchange.config.MetadataExchange",
            "protocol": "istio-peer-exchange"
           }
          },
```

We confirmed that the env var is `PILOT_ENABLE_METADATA_EXCHANGE:                  false` in istiod pods.

### Version

```prose
Works in Istio 1.11.4
Broke in Istio 1.12.6
```

**Comment by zirain** [comment]
can you share the output of `kubectl get envoyfilter -nistio-system`?

**Comment by weibohe** [comment]
@zirain - I can confirm that we don't have any EnvoyFilter that inserts "istio.metadata_exchange"

**Comment by zirain** [comment]
seems related to this https://github.com/istio/istio/pull/36756, need to verify.

**Comment by ramaraochavali** [comment]
We can. Are you running in to any issue because of this?

**Comment by weibohe** [comment]
We found this during upgrade so did not have production impact. We have found a way to work around this - but it's appreciated if we can back-port the fix into v1.12!
